packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -0,0 +1,257 @@
+// @flow
+import asyncMap from '@xen-orchestra/async-map'
+import defer from 'golike-defer'
+import { pipeline } from 'readable-stream'
+import { promisify, ignoreErrors } from 'promise-toolbox'
+
+import { type Xapi } from '../xapi'
+import {
+  safeDateFormat,
+  type SimpleIdPattern,
+  unboxIdsFromPattern,
+} from '../utils'
+
+import { type Executor, type Job } from './jobs'
+import { type Schedule } from './scheduling'
+
+const pPipeline = promisify(pipeline)
+
+const METADATA_BACKUP_JOB_TYPE = 'metadataBackup'
+
+type Settings = {|
+  retentionXoMetadata?: number,
+  retentionPoolMetadata?: number,
+|}
+
+type MetadataBackupJob = {
+  ...$Exact<Job>,
+  pools?: SimpleIdPattern,
+  remotes: SimpleIdPattern,
+  settings: $Dict<Settings>,
+  type: METADATA_BACKUP_JOB_TYPE,
+  xoMetadata?: boolean,
+}
+
+const METADATA_BACKUP_DIR = 'xo-metadata-backups'
+
+// File structure on remotes:
+//
+// <remote>
+// └─ xo-metadata-backups
+//   └─ <schedule ID>
+//    ├─ xo
```

Comment:
What if multiple XOs save their config on this remote?

Reply:
The folder will be override. Is the XO instance has an id or a thing that identify it to another XO?

Reply:
I don't think so.

@pdonias Any ideas?

Reply:
I'm not sure to understand the question. The <schedule ID> folder should prevent any override anyway but why would anything be overridden, exactly?

However, the cohabitation may become an issue once the config restoration is implemented because it will show the config backups of all the XOs, which is kind of weird.

Reply:
Right, my bad the folder will not be overridden but we cannot differentiate The XOs. The question is if the XO instance has a property or a think that help we to identify.

Reply:
I think we should let the user name its XO, otherwise how will he knows which config to restore?

Reply:
Right, I think i should add a field in the UI to provide it.

Reply:
The issue with my suggestion is if a user edit the name it not means that the backupped metadata refers to a different XO.